libminifi/include/core/PropertyValidation.h

```diff
@@ -259,6 +259,22 @@ class UnsignedLongValidator : public PropertyValidator {
   }
 };
 
+class NonBlankValidator : public PropertyValidator {
+ public:
+  explicit NonBlankValidator(const std::string& name)
+      : PropertyValidator(name) {
+  }
+  ~NonBlankValidator() override = default;
+
+  ValidationResult validate(const std::string& subject, const std::shared_ptr<minifi::state::response::Value>& input) const final {
+    return validate(subject, input->getStringValue());
```

Comment:
How would it validate not being blank?
I think we should trim both left and right and check if the result is empty.

Reply:
I thought yaml parsing ignored padding, so
```yaml
name:  A
```
was the same as 
```yaml
name: A
```